test/boost/database_test.cc

```diff
@@ -409,33 +409,42 @@ SEASTAR_THREAD_TEST_CASE(test_distributed_loader_with_pending_delete) {
 
     const sstring toc_text = "TOC.txt\nData.db\n";
 
+    std::optional<sstables::generation_type> prev_gen;
+    std::vector<sstables::generation_type> gen;
+    size_t num_gens = 9;
+    gen.reserve(num_gens);
```

Comment:
nit, could use `std::generate_n()` so we don't need to leave `prev_gen` in this scope. will create a follow-up change to address this once this change is landed.